packages/input/src/calcTextareaHeight.js

```diff
@@ -97,6 +100,11 @@ export default function calcTextareaHeight(
     }
     height = Math.min(maxHeight, height);
   }
+
+  if (boxSizing === 'border-box') {
+    height = Math.round((height - paddingSize) / lineheight) * lineheight + paddingSize + borderSize;
```

Comment:
If we go with this line-height route, just put a comment that explains what is going on to the best of your ability. I know we don't fully understand it.